src/ActiveBackground.tsx

```diff
@@ -1,18 +1,34 @@
-import * as React from 'react'
+import type { ReactNode } from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
 import { useResizeObserver } from './hooks/useResizeObserver'
-import { IActiveBackground, IActiveBackgroundPattern } from './index'
 
-export const ActiveBackground: React.FC<IActiveBackground> = ({
-  pattern,
-  patternOptions = {},
-  className,
-  children,
-}) => {
-  const [canvasRef, setRef] = React.useState<HTMLCanvasElement | null>(null)
-  const onRefSet = React.useCallback((ref) => setRef(ref), [setRef])
-  const [update, setUpdate] = React.useState({})
+export type PatternOptions = Record<string, unknown>
+export interface PatternConstructor<TOptions> {
+  new (
+    canvas: HTMLCanvasElement,
+    options?: TOptions
+  ): ActiveBackgroundPattern
+}
+export interface ActiveBackgroundPattern {
+  render(): void
+  start(): void
+  stop(): void
+}
+
+export interface ActiveBackgroundProps<TPatternOptions = PatternOptions> {
+  Pattern: PatternConstructor<TPatternOptions>
```

Comment:
Yes, you can case your constructors `PascalCase`. Not a necessity, but I like the visual difference.